init/lib/status_test.go

```diff
@@ -14,28 +14,56 @@
 
 package lib
 
-//func TestIsRunning(t *testing.T) {
-//	assert.False(t, isRunning(99999))
-//
-//	myPid := os.Getpid()
-//	assert.True(t, isRunning(myPid))
-//}
-//
-//func TestIsRunningByPidFile(t *testing.T) {
-//	running, err := IsRunningByPidFile("bogus file")
-//	require.Error(t, err)
-//	assert.EqualError(t, err, "open bogus file: no such file or directory")
-//	assert.Equal(t, running, 3)
-//
-//	assert.NoError(t, ioutil.WriteFile("pidfile", []byte("99999"), 0644))
-//	running, err = IsRunningByPidFile("pidfile")
-//	require.NoError(t, err)
-//	assert.Equal(t, running, 1)
-//
-//	assert.NoError(t, ioutil.WriteFile("pidfile", []byte(strconv.Itoa(os.Getpid())), 0644))
-//	running, err = IsRunningByPidFile("pidfile")
-//	require.NoError(t, err)
-//	assert.Equal(t, running, 0)
-//
-//	assert.NoError(t, os.Remove("pidfile"))
-//}
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"fmt"
+)
+
+func TestIsRunning_Running(t *testing.T) {
+	process, _ := os.FindProcess(os.Getpid())
+	assert.True(t, isRunning(process))
+}
+
+func TestIsRunning_NotRunning(t *testing.T) {
+	process, _ := os.FindProcess(99999)
+	assert.False(t, isRunning(process))
+}
+
+func TestGetProcessStatus_Running(t *testing.T) {
+	setup()
+	defer Teardown()
+
+	WritePid(os.Getpid())
+	process, status, err := GetProcessStatus()
+
+	assert.Equal(t, process.Pid, os.Getpid())
+	assert.Equal(t, status, 0)
+	assert.NoError(t, err)
+}
+
+func TestGetProcessStatus_NotRunningPidfileExists(t *testing.T) {
+	setup()
+	defer Teardown()
+
+	notRunningPid := 99999
+	WritePid(notRunningPid)
+	process, status, err := GetProcessStatus()
+
+	assert.Equal(t, process.Pid, notRunningPid)
+	assert.Equal(t, status, 1)
+	assert.EqualError(t, err, "pidfile exists but process is not running")
+}
+
+func TestGetProcessStatus_NotRunningPidfileDoesNotExist(t *testing.T) {
+	setup()
+	defer Teardown()
+
+	process, status, err := GetProcessStatus()
+
+	assert.Empty(t, process)
+	assert.Equal(t, status, 3)
+	msg := fmt.Sprintf("failed to read pidfile: open %s: no such file or directory", Pidfile)
+	assert.Contains(t, err.Error(), msg)
```

Comment:
The `assert` and `require` functions take format strings, so no need to create `msg` separately:

```go
assert.Contains(t, err.Error(), "failed to read pidfile: open %s: no such file or directory", Pidfile)
```

Reply:
Yeah, it doesn't look like this is the case unless I'm misreading something:

```
func Contains(t TestingT, s, contains interface{}, msgAndArgs ...interface{}) bool {
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}

	ok, found := includeElement(s, contains)
...
```
I think the error message if it fails can be a format string, but not the actual `contains` since it's untyped

Reply:
@gregakinman you are correct, the formatting is only for a message to produce on the assertion failing, not on the thing to check, so you were correct the first time.